src/compute/src/logging/mod.rs

```diff
@@ -155,3 +156,52 @@ struct SharedLoggingState {
     /// Shared compute logger.
     compute_logger: Option<ComputeLogger>,
 }
+
+/// Helper to pack collections of [`Datum`]s into key and value row.
+pub(crate) struct PermutedRowPacker {
+    key_row: Row,
+    value_row: Row,
+    key: Vec<usize>,
+    value: Vec<usize>,
+}
+
+impl PermutedRowPacker {
+    /// Construct based on the information within the log variant.
+    pub(crate) fn new(variant: LogVariant) -> Self {
```

Comment:
You could get rid of the `.into()`s at the call site and avoid some noise there by changing this to an `fn new<V: Into<LogVariant>>(variant: V) -> Self`.

Reply:
That's nicer! Thanks.